src/backend/api/Fusion.Resources.Logic/Requests/Commands/ResourceAllocationRequest/Allocation/InitializedHandler.cs

```diff
@@ -28,24 +26,45 @@ public InitializedHandler(ResourcesDbContext dbContext, IMediator mediator)
 
                 public async Task Handle(RequestInitialized notification, CancellationToken cancellationToken)
                 {
-                    if (notification.Type != DbInternalRequestType.Direct)
+                    if (notification.Type != DbInternalRequestType.Allocation)
                         return;
 
                     var initiatedBy = await dbContext.Persons.FirstAsync(p => p.Id == notification.InitiatedByDbPersonId);
                     var request = await dbContext.ResourceAllocationRequests.FirstAsync(r => r.Id == notification.RequestId);
 
+                    var subType = notification.SubType;
+
+                    if (subType is null)
+                    {
+                        if (request.OrgPositionId is null)
+                            throw new InvalidOperationException("Org position id is null for request. Cannot resolve sub type.");
+
+                        subType = await mediator.Send(new ResolveSubType(request.OrgPositionId.Value, request.OrgPositionInstance.Id));
+                    }
+
+                    WorkflowDefinition workflow = subType.ToLower() switch
+                    {
+                        AllocationDirectWorkflowV1.SUBTYPE => new AllocationDirectWorkflowV1(initiatedBy),
+                        AllocationJointVentureWorkflowV1.SUBTYPE => new AllocationJointVentureWorkflowV1(initiatedBy),
+                        AllocationNormalWorkflowV1.SUBTYPE => new AllocationNormalWorkflowV1(initiatedBy),
+                        _ => throw new NotSupportedException($"Sub type '{subType}' is not supported for initialization.")
+                    };
 
-                    var workflow = new InternalRequestDirectWorkflowV1(initiatedBy);
                     dbContext.Workflows.Add(workflow.CreateDatabaseEntity(notification.RequestId, DbRequestType.InternalRequest));
 
                     request.LastActivity = DateTime.UtcNow;
-                    request.State.State = InternalRequestDirectWorkflowV1.CREATED;
+                    request.State.State = AllocationDirectWorkflowV1.CREATED;
+
+                    // Try to resolve the default assigned department
+                    request.AssignedDepartment = await mediator.Send(new Queries.ResolveResponsibleDepartment(request.Id));
```

Comment:
Would this not overwrite the assigned department if it is sent through the API, or are we deprecating that option?

Reply:
It is sort of deprecated, but could add a nullcheck here, so we do not blindly overwrite..